zilliqa/src/bin/validator_oracle.rs

```diff
@@ -0,0 +1,238 @@
+use std::{path::PathBuf, str::FromStr};
+
+use alloy_provider::{Provider, ProviderBuilder, WsConnect};
+use alloy_rpc_types::{BlockNumberOrTag, Filter};
+use anyhow::Result;
+use clap::Parser;
+use ethers::{
+    contract::abigen,
+    providers::Middleware,
+    signers::{LocalWallet, Signer},
+    types::{TransactionRequest, H160},
+};
+use futures_util::stream::StreamExt;
+use tokio::sync::watch;
+use tracing::{debug, error, info};
+use tracing_subscriber::EnvFilter;
+use zilliqa::{
+    contracts,
+    crypto::{NodePublicKey, SecretKey},
+    state::contract_addr,
+    uccb::{
+        cfg::{ChainConfig, Config, ZQ2Config},
+        client::ChainClient,
+    },
+};
+
+// The json files are generated by invoking 'forge build' in zilliqa/src/contracts/uccb
+abigen!(
+    ValidatorManager,
+    "zilliqa/src/contracts/uccb/out/ValidatorManager.sol/ValidatorManager.json"
+);
+
+#[derive(Parser, Debug)]
+struct Args {
+    #[arg(value_parser = SecretKey::from_hex)]
+    secret_key: SecretKey,
+    #[clap(long, short, default_value = "config.toml")]
+    config_file: PathBuf,
+}
+
+impl zilliqa::uccb::Args for Args {
```

Comment:
Here I could, but this will be used in zilliqa as well to specify the UCCB config file where config_file will point to the zq2 one instead of the UCCB one.